feature(cache): format freed bytes for readability

Alizter opened the pull request.

snowleopard commented on bin/cache.ml:
```diff
@@ -35,8 +35,16 @@ let trim ~trimmed_size ~size =
   with
   | Error s -> User_error.raise [ Pp.text s ]
   | Ok { trimmed_bytes } ->
+    let pp_bytes trimmed_bytes =
```
I think this code should live somewhere in Stdune as it's generally useful.

snowleopard replied:
(In fact, don't we already have something similar?)

Alizter replied:
I couldn't find anything about formatting bytes like this in the repo. I can add it to stdune, though I am not sure who else will use it. Where should I include it?

emillon replied:
There's `bytes_unit` in `Decoder` (`dune_sexp`). You might be able to reuse the table.

snowleopard replied:
Ha, indeed! Thanks @emillon. I remember seeing something like this -- it even has a comment from me on top :)

Alizter replied:
I've factored the parsing and printing of byte units through a library in stdune called Bytes_units. This is backwards compatible and allows aliases for units to easily be added.

snowleopard replied:
Thanks!

snowleopard commented on otherlibs/stdune/bytes_unit.ml:
```diff
@@ -0,0 +1,37 @@
+  else
+    Printf.sprintf "%.2f %s" (Int64.to_float x /. Int64.to_float value) suffix
+
+let () =
```
It seems weird to test by running code at the top level. Can we turn this into a proper test that runs only as part of the test suite, not on every execution of Dune?

Alizter replied:
I've moved it into an expect test. Not sure what I was thinking when I wrote it.

snowleopard commented on otherlibs/stdune/bytes_unit.ml:
```diff
@@ -0,0 +1,37 @@
+let conversion_table =
+  [ ([ "bytes"; "B" ], 1L)
+  ; ([ "KB"; "kB"; "KiB"; "kilobytes" ], 1000L)
```
`KiB` is not 1000, it's 1024. Using `MiB` and `GiB` below isn't accurate either.

snowleopard reviewed (changes requested):
Great job with the comments! I left a couple of suggestions.

By the way, it seems weird that no test changed in the process. I guess the reason is that the only test we have continues to use "bytes". I guess it's fine.

Alizter commented:
I was careful not to change the tests before, but now I have changed it from `bytes` to `B` to be consistent. I've added an expect test making sure the edge cases are handled correctly.

I am not going to add support for MiB, I actually never knew about MiB before and it was an extremely confusing read. If I add it in our suggested units helper messages will also display them etc. so it looks too tricky to be worth it.

snowleopard commented on test/blackbox-tests/test-cases/dune-cache/trim.t:
```diff
@@ -34,7 +34,7 @@ Function to reset build tree and cache.
 
   $ dune cache trim --size 0B
-  Freed 0 bytes
+  Freed 0 B
```
The space between 0 and B seems unnecessary.

snowleopard replied:
(We usually write just "5GB".)

Alizter replied:
I've removed the space.

snowleopard reviewed (approved):
Looks great, thank you!

snowleopard commented:
> I am not going to add support for MiB, I actually never knew about MiB before and it was an extremely confusing read. If I add it in our suggested units helper messages will also display them etc. so it looks too tricky to be worth it.

Yeah, I'm also not sure it's worth it.
